OneCommunityGlobal/HGNRest
# Yi - Add OverviewReport APIs

# Description
<img width="743" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/59299011/63131e45-d54d-419f-b98f-b160f200198a">


## Related PRS (if any):
Frontend PR is working in progress.

## Main changes explained:
- Add `getBlueStats` API, given start and end date, it returns the number of BlueStats assigned fall between the date.
- Add `getVolunteerRoleStats` API, given start and end date, it returns 
  - 4+ members team count
  - Total badges awarded count
  - Number of users celebrating their anniversary
  - Role and count of users 
- Add `getVolunteerHoursStats` API, given start and end date, it returns total number of users and group them by hours logged.
- Add `getVolunteerStats` API, given start and end date, it returns 
  - 4+ members team count
  - Total badges awarded count
  - Number of users celebrating their anniversary
  - teamMembersCount
  - activeInactiveUsersCount
- Add `getTaskProjectStats` API, given start and end date, it returns 
  - Total hours logged in tasks
  - Total hours logged in projects
  - Number of member with tasks assigned
  - Number of member without tasks assigned
  - Number of tasks with due date within the date range

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. test `getBlueStats` API:
  - `/api/reports/overviewsummaries/bluestats?startDate=&endDate=`
4. test `getVolunteerRoleStats` API:
  - `/api/reports/overviewsummaries/volunteerrolestats?startDate=&endDate=`
5. test `getVolunteerHoursStats` API: 
  - `/api/reports/overviewsummaries/volunteerhoursstats?startDate=&endDate=`
6. test `getVolunteerStats` API: 
  - `/api/reports/overviewsummaries/volunteerstats?startDate=&endDate=`
7. test `getTaskProjectStats` API: 
  - `/api/reports/overviewsummaries/taskandprojectstats?startDate=&endDate=`

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HGNRest/assets/59299011/e5364ccd-89ef-4675-8cc5-48d32e6f5028


https://github.com/OneCommunityGlobal/HGNRest/assets/59299011/7b5933a4-091e-49b4-883a-8112bee55924


https://github.com/OneCommunityGlobal/HGNRest/assets/59299011/462dc735-ebfe-4119-a531-18e90248565d
